src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -155,6 +159,38 @@ public void close() {
     }
   }
 
+  /**
+   * Determine the acceptable means of authentication for a given DRS ID, including the passport
+   * issuers when supported.
+   *
+   * @param drsObjectId the object ID for which to look up authorizations
+   * @return the `DrsAuthorizations` for this ID
+   * @throws IllegalArgumentException if there is an issue with the object id
+   * @throws SnapshotNotFoundException if the snapshot for the DRS object cannot be found
+   * @throws TooManyRequestsException if there are too many concurrent DRS lookup requests
+   */
+  public DRSAuthorizations lookupAuthorizationsByDrsId(String drsObjectId) {
+    try (DrsRequestResource r = new DrsRequestResource()) {
+      DRSAuthorizations auths = new DRSAuthorizations();
+
+      auths.addSupportedTypesItem(DRSAuthorizations.SupportedTypesEnum.BEARERAUTH);
+      // TODO: add to bearer_auth_issuers, ask Muscles.
+
+      SnapshotCacheResult snapshot = lookupSnapshotForDRSObject(drsObjectId);
+      SnapshotSummaryModel snapshotSummary = getSnapshotSummary(snapshot.id);
```

Comment:
I'm thinking that the consent code should live in the `SnapshotCacheResult` cache too (since I don't think it changes very frequently and this `getSnapshotSummary` requires a call to the DB and I'd worry that this could add a fair amount of perf overhead)

Reply:
Ah!  Good shout, I agree.

Reply:
@nmalfroy do you think we should also cache the PHS ID?

As an aside, both the snapshot's PHS ID and consent code were already fetched as part of the existing GetObject implementation, but I still agree that caching them makes sense.

Reply:
Ah, great point!  Yeah, I'd definitely cache both

Reply:
Circling back here again:

Locally I refactored out the snapshot summary and moved the PHS ID and consent code to the `SnapshotCacheResult`.  But before committing I looked more closely at the existing code, it looks like snapshot summaries were also being stored in their own cache, so now I don't think this is needed.  What do you think?

https://github.com/DataBiosphere/jade-data-repo/blame/develop/src/main/java/bio/terra/service/filedata/DrsService.java#L579-L581
https://github.com/DataBiosphere/jade-data-repo/pull/1252/commits/3ea82bbf7dc575ba4bedb523aa422db656e4fbf3#diff-e0de83588914b4facd07a2e3f7f6c9c3d52b613924d2cb4e94c53570975b5c5d

Reply:
Thanks for checking!

> Do you mind addressing the logic (drsservice vs. snapshot summary thing)

This was @tlangs request, right?  Working on it now!

https://github.com/DataBiosphere/jade-data-repo/pull/1266#discussion_r857661604